llvm/lib/CodeGen/RegisterCoalescer.cpp

```diff
@@ -3672,6 +3672,14 @@ bool RegisterCoalescer::joinVirtRegs(CoalescerPair &CP) {
     // having stale segments.
     LHSVals.pruneMainSegments(LHS, ShrinkMainRange);
 
+    LHSVals.pruneSubRegValues(LHS, ShrinkMask);
+    RHSVals.pruneSubRegValues(LHS, ShrinkMask);
+  } else if (TrackSubRegLiveness && !CP.getDstIdx() && CP.getSrcIdx()) {
+    LHS.createSubRangeFrom(LIS->getVNInfoAllocator(),
+                           CP.getNewRC()->getLaneMask(), LHS);
+    mergeSubRangeInto(LHS, RHS, TRI->getSubRegIndexLaneMask(CP.getSrcIdx()), CP,
+                      CP.getDstIdx());
+    LHSVals.pruneMainSegments(LHS, ShrinkMainRange);
     LHSVals.pruneSubRegValues(LHS, ShrinkMask);
     RHSVals.pruneSubRegValues(LHS, ShrinkMask);
```

Comment:
Looks no need to prune `RHS`'s subregvals, since RHS doesn't contain any subranges.

Reply:
I'm not sure what you mean by RHS doesn't contain any subregs. I think I remember seeing subregs on both the RHS and LHS in IR. Taken from one of our tests for example:
```
%48:vsrprc = INSERT_SUBREG %45, killed %47, %subreg.sub_vsx0
```
This is probably not what you mean...

Reply:
> I'm not sure what you mean by RHS doesn't contain any subregs.

`RHS` as a LiveInterval doesn't contain any subranges(if it does, another branch prior this one has handled that). Looking into `pruneSubRegValues`, the method will iterate over LiveInterval's subranges and shrink range accordingly. Since `RHS` doesn't contain any subranges, so `RHS` should not be changed.